src/rocketsonde/core/probe.py

```diff
@@ -55,30 +67,26 @@ def probe_pid(self):
     def start_monitor(self):
         pid_queue = Queue(100)
         record_queue = Queue(10000)
-        shutdown_event = Event()
-        records_ready = Event()
-        process = Process(
-            target=self._worker_function, args=(self._metric, pid_queue, record_queue, records_ready, shutdown_event)
-        )
+        process = Process(target=self._worker_function, args=(self._metric, pid_queue, record_queue))
 
         process.start()
         atexit.register(self._panic_kill_monitor)
 
-        self._worker = _Worker(
-            shutdown_event=shutdown_event,
-            pid_queue=pid_queue,
-            record_queue=record_queue,
-            records_ready=records_ready,
-            process=process,
-        )
+        self._worker = _Worker(pid_queue=pid_queue, record_queue=record_queue, process=process)
 
     def _drain_record_pipe(self):
         if self._worker is not None:
-            while self._worker.records_ready.is_set() or not self._worker.record_queue.empty():
+            pipe_clean = False
+            while (self._worker.process.is_alive() and not pipe_clean) or not self._worker.record_queue.empty():
```

Comment:
This complexity is difficult to debug. The two calls, is_alive() and queue.empty() can be set or unset within the span of this line.

Reply:
I don't think this region needs to be tight, it's just there to dump out of the loop if the worker is gone and can no longer send the ALL_WRITTEN message. But, as you say above, this would be easier to test if it were written out formally. So that's a thing to do.

Reply:
And, again. I thought there was more risk in this part, but there isn't.